iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RevertConnectVoltageLevelOnLine.java

```diff
@@ -0,0 +1,187 @@
+import static com.powsybl.iidm.modification.topology.ModificationReports.noVoltageLevelInCommonReport;
+import static com.powsybl.iidm.modification.topology.ModificationReports.notFoundLineReport;
+import static com.powsybl.iidm.modification.topology.ModificationReports.removedLineReport;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.LoadingLimitsBags;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.addLoadingLimits;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.attachLine;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.createLineAdder;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.removeVoltageLevelAndSubstation;
+
+/**
+ * This method reverses the action done in the ConnectVoltageLevelOnLine class :
+ * it replaces 2 existing lines (with the same voltage level at one of their side) with a new line,
+ * and eventually removes the voltage level in common (switching voltage level), if it contains no equipments anymore, except bus or bus bar section
+ *
+ *    VL1 ----------- switching voltage level ----------- VL2         =========>        VL1 ------------------------- VL2
+ *          (line1)                             (line2)                                              (line)
+ *
+ *
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+public class RevertConnectVoltageLevelOnLine extends AbstractNetworkModification {
+
+    private String line1Id;
+    private String line2Id;
+
+    private String lineId;
+    private String lineName;
+
+    /**
+     * Constructor.
+     *
+     * @param line1Id       The non-null ID of the first line
+     * @param line2Id       The non-null ID of the second line
+     * @param lineId        The non-null ID of the new line to be created
+     * @param lineName      The optional name of the new line to be created
+     */
+    RevertConnectVoltageLevelOnLine(String line1Id, String line2Id, String lineId, String lineName) {
+        this.line1Id = Objects.requireNonNull(line1Id);
+        this.line2Id = Objects.requireNonNull(line2Id);
+        this.lineId = Objects.requireNonNull(lineId);
+        this.lineName = lineName;
+    }
+
+    public RevertConnectVoltageLevelOnLine setLine1Id(String line1Id) {
+        this.line1Id = Objects.requireNonNull(line1Id);
+        return this;
+    }
+
+    public RevertConnectVoltageLevelOnLine setLine2Id(String line2Id) {
+        this.line2Id = Objects.requireNonNull(line2Id);
+        return this;
+    }
+
+    public RevertConnectVoltageLevelOnLine setLineId(String lineId) {
+        this.lineId = Objects.requireNonNull(lineId);
+        return this;
+    }
+
+    public RevertConnectVoltageLevelOnLine setLineName(String lineName) {
+        this.lineName = lineName;
+        return this;
+    }
+
```

Comment:
```
boolean failsOrThrowLinePresence(Network network, String lineId, Reporter reporter, boolean throwException) {
   Line line = network.getLine(lineId);
        if (line == null) {
            notFoundLineReport(reporter, lineId);
            if (throwException) {
                throw new PowsyblException(String.format("Line %s is not found", line1Id));
            }
            return true;
        }
    return false;
}

// [...]

if (failsOrThrowLinePresence(network, line1Id, reporter, throwException)
  || failsOrThrowLinePresence(network, line2Id, reporter, throwException)) {
   return;
}
```
Without our choice of indentation rules and mandatory braces, I would have opted for two separated `if () return;`

Reply:
Code smell concerning complexity has been solved